### PR TITLE
Add response=sealed|plain mode for OpenAPI client generation

### PR DESCRIPTION
**Pull request by dsudomoin**
## Summary

- New `response=sealed|plain` generator parameter for HTTP client code generation
- In `plain` mode, client methods return the success type directly (e.g. `List<Pet>`) instead of sealed response wrappers
- Non-2xx responses throw `HttpClientResponseException` with deserialized typed body when error schema is defined in spec
- Automatic fallback to `sealed` when multiple 2xx responses have different types
- Works across all 5 client modes: Java sync/async/reactive, Kotlin blocking/suspend

## Changes

- **`HttpClientResponseException`** — added `@Nullable Object body` field, `of()` / `ofRaw()` factory methods, typed `body()` / `body(Class)` accessors
- **`KoraCodegen.java`** — `ResponseStyle` enum (SEALED/PLAIN), CLI option `response`, vendor extension logic with fallback in `postProcessOperationsWithModels()`
- **Java templates** — `javaApiResponses`, `javaClientApi`, `javaClientResponseMappers`, `javaClientAsyncResponseMappers` updated with conditional plain response blocks
- **Kotlin templates** — `kotlinApiResponses`, `kotlinClientApi`, `kotlinClientResponseMappers`, `kotlinClientAsyncResponseMappers` updated similarly
- **Tests** — new `petstoreV3_plain_response.yaml` spec covering all edge cases, 15 new test cases (405 total, 0 failures)

## Test plan

- [x] Plain mode generates correct API interface with direct return types
- [x] Plain mode generates PlainResponseMapper with per-code error handling
- [x] Fallback to sealed for multiple 2xx with different types (multiResponse operation)
- [x] No-body 204 returns `Void`/`Unit`
- [x] Multiple 2xx same type stays in plain mode
- [x] All 5 client modes compile successfully
- [x] All existing sealed-mode tests pass (no regressions)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: http/http-client-common/src/main/java/ru/tinkoff/kora/http/client/common/HttpClientResponseException.java [hunk 1]
```diff
@@ -12,12 +13,43 @@ public class HttpClientResponseException extends HttpClientException {
     private final int code;
     private final HttpHeaders headers;
     private final byte[] bytes;
+    @Nullable
+    private final Object body;
```
[the remark this answers is not in the document]
Comment (dsudomoin) [reply http/http-client-common/src/main/java/ru/tinkoff/kora/http/client/common/HttpClientResponseException.java]: Разделил на два эксепшна:
- `HttpClientResponseException` — оригинальный, с raw bytes (code, headers, bytes)
- `HttpClientDecodedResponseException extends HttpClientResponseException` — с типизированным body и аксессорами `body()` / `body(Class<T>)`

Шаблоны обновлены — для decoded ошибок кидается `HttpClientDecodedResponseException`, для fallback без схемы — `HttpClientResponseException.ofRaw()`.

**Comment by dsudomoin** [comment]
Closing in favor of new PR targeting master with rewritten implementation for 2.0.0 architecture (JavaPoet/KotlinPoet generators instead of Mustache templates).
